crates/hir-ty/src/db.rs

```diff
@@ -107,6 +108,12 @@ pub trait HirDatabase: DefDatabase + Upcast<dyn DefDatabase> {
     #[salsa::invoke(crate::layout::target_data_layout_query)]
     fn target_data_layout(&self, krate: CrateId) -> Result<Arc<TargetDataLayout>, Arc<str>>;
 
+    #[salsa::invoke(crate::object_safety::object_safety_query)]
+    fn object_safety(&self, trait_: TraitId) -> Option<ObjectSafetyViolation>;
+
+    #[salsa::invoke(crate::object_safety::object_safety_of_trait_query)]
+    fn object_safety_of_trait(&self, trait_: TraitId) -> Option<ObjectSafetyViolation>;
```

Comment:
Why do we have two of these?

Reply:
I was trying to avoid supertrait traversals so I've made the first one, but I think that if that might be the bottle neck, making supertraits into query is more reasonable 😅 Anyway, the first one seems redundant